mrjob/spark/mrjob_spark_harness.py

```diff
@@ -72,17 +72,43 @@ def _run_step(step, step_num, rdd, make_job):
             lambda pairs: mapper_job.map_pairs(pairs, step_num))
         rdd = rdd.map(lambda k_v: m_write(*k_v))
 
+    if step_desc.get('combiner'):
+        combiner_job = make_job('--combiner', '--step-num=%d' % step_num)
+        c_read, c_write = combiner_job.pick_protocols(step_num, 'combiner')
+
+        # run the combiner
+        rdd = rdd.map(c_read)
+
+        def combiner_helper(pairs1, pairs2):
+            if len(pairs1) == len(pairs2) == 1:
+                return list(
+                    combiner_job.combine_pairs(pairs1 + pairs2, step_num),
+                )
+            else:
+                pairs1.extend(pairs2)
+                return pairs1
+
+        rdd = rdd.map(lambda k_v: (k_v[0], k_v))
```

Comment:
This would be a good place for a comment explaining that mrjob combiners might potentially behave differently for different keys, so we need to include the key in the "value"